modules/local/tree_summary.nf

```diff
@@ -0,0 +1,50 @@
+process TREE_SUMMARY {
+    tag "$meta.id"
+    label 'process_single'
+
+    container = 'ecoflowucl/genomeqc_tree:v1.3'
+    publishDir "$params.outdir/tree_plots" , mode: "${params.publish_dir_mode}", pattern:"Phyloplot_*.pdf"
+
+    input:
+    tuple val(meta), path(tree)
+    path  multiqc_files
+
+    output:
+    path( "P*.pdf"          ),                emit: figure
+    path( "versions.yml"    ),                emit: versions
+
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+
+    #Remove unwanted extensions in the tree file
+    sed \'s/\\.prot\\.fa\\.largestIsoform//g\' ${tree}/Species_Tree/SpeciesTree_rooted_node_labels.txt > tree.nw
```

Comment:
This might be easier to do in groovy.

files have `getSimpleName` and `getBaseName(int)` functions to strip file endings. There's also the `-` operation on strings that will strip a string.